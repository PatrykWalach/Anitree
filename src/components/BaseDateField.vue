<template>
  <v-menu
    ref="menu"
    v-model="menuActive"
    :close-on-content-click="false"
    :return-value.sync="date"
    transition="scale-transition"
    offset-y
    min-width="290px"
  >
    <template v-slot:activator="{ on, attrs }">
      <v-text-field
        v-model="date"
        prepend-inner-icon="event"
        readonly
        v-bind="{ ...attrs, ...$attrs }"
        v-on="on"
        @click:clear="clear"
      ></v-text-field>
    </template>
    <v-date-picker v-model="date" no-title scrollable>
      <v-spacer></v-spacer>
      <v-btn text color="primary" @click="menuActive = false">Cancel</v-btn>
      <v-btn text color="primary" @click="save">OK</v-btn>
    </v-date-picker>
  </v-menu>
</template>
<script lang="ts">
import { Ref, createComponent, ref, watch } from '@vue/composition-api'

export interface Props {
  value: string
}

export default createComponent<Readonly<Props>>({
  inheritAttrs: false,
  props: {
    value: {
      default: '',
      required: true,
      type: String,
    },
  },
  setup(props, { emit }) {
    const date = ref('')
    const menuActive = ref(false)
    const menu: Ref<any> = ref(null)

    watch(
      () => props.value,
      value => {
        date.value = value
      },
    )

    const save = () => {
      const newDate = date.value

      menu.value.save(newDate)
      if (newDate !== props.value) {
        emit('change', newDate)
      }
    }

    const clear = () => {
      emit('change', '')
    }

    return { clear, date, menu, menuActive, save }
  },
})
</script>
