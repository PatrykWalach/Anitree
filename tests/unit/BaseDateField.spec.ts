import { createLocalVue, shallowMount } from '@vue/test-utils'
import BaseDateField from '@/components/BaseDateField.vue'
import VueCompostionApi from '@vue/composition-api'
import Vuetify from 'vuetify'

describe('BaseDateField', () => {
  const localVue = createLocalVue()
  localVue.use(Vuetify)
  localVue.use(VueCompostionApi)
  test('matches snapshot', () => {
    const wrapper = shallowMount(BaseDateField, {
      localVue,
      propsData: {
        value: '',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
