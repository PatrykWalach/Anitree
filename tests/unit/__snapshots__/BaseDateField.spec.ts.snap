// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDateField matches snapshot 1`] = `
<v-menu-stub
  closedelay="0"
  closeonclick="true"
  contentclass=""
  maxheight="auto"
  maxwidth="auto"
  minwidth="290px"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  offsety="true"
  opendelay="0"
  openonclick="true"
  origin="top left"
  returnvalue=""
  transition="scale-transition"
>
   
  <v-date-picker-stub
    eventcolor="warning"
    firstdayofweek="0"
    nexticon="$next"
    notitle="true"
    previcon="$prev"
    scrollable="true"
    selecteditemstext="$vuetify.datePicker.itemsSelected"
    showcurrent="true"
    type="date"
    value=""
    width="290"
  >
    <v-spacer-stub />
     
    <v-btn-stub
      activeclass=""
      color="primary"
      tag="button"
      text="true"
      type="button"
    >
      Cancel
    </v-btn-stub>
     
    <v-btn-stub
      activeclass=""
      color="primary"
      tag="button"
      text="true"
      type="button"
    >
      OK
    </v-btn-stub>
  </v-date-picker-stub>
</v-menu-stub>
`;
